api/tests/mocha/services/authorization.spec.js

```diff
@@ -254,6 +353,107 @@ describe('Authorization service', () => {
+              { id: 'r2', key: 'contact', value: { type: 'data_record' } }, // depth 1
+              { id: 'r3', key: 'parent', value: { type: 'task' } }, // not a report, but depth 0
+              { id: 'r4', key: 'contact', value: { type: 'target' } },  // not a report, but depth 1
+              { id: 'r5', key: 'parent', value: { type: 'contact' } },  // not a report, but depth 0
+              { id: 'r6', key: 'subject', value: { type: 'data_record', submitter: 'some_person' } }, // depth 2
+              { id: 'r7', key: 'contact', value: { type: 'data_record', submitter: 'some_person' } }, // depth 1
+              { id: 'r8', key: 'subject', value: { type: 'target' } },  // not a report, but depth 2
+              { id: 'r9', key: 'subject', value: { type: 'data_record', submitter: 'contact_id' } }, // depth 2, self
+            ]
+        });
+
+      return service
+        .getAllowedDocIds({
+          subjectIds,
+          userCtx: { name: 'user', facility_id: 'facility_id', contact_id: 'contact_id' },
+          contactDepth: 2,
+          reportDepth: 1,
+          subjectsDepth: { 'parent': 0, 'contact': 1, 'subject': 2 },
+        })
+        .then(result => {
+          result.should.have.members([
+            '_design/medic-client', 'org.couchdb.user:user',
+            'r2', 'r3', 'r4', 'r5', 'r7', 'r8', 'r9'
+          ]);
+        });
+    });
+
+    it('should check all entries for a report to verify valid depth', () => {
+      const subjectIds = ['contact', 'parent', 'place'];
+      db.medic.query
+        .withArgs('medic/docs_by_replication_key')
+        .resolves({ rows:
+            [
+              { id: 'r1', key: 'place', value: { submitter: 'p', type: 'data_record', id: 'r1'} }, // depth 1
+              { id: 'r1', key: 'parent', value: { submitter: 'p', type: 'data_record', id: 'r1' } }, // depth 0
+              { id: 'r2', key: 'place', value: { submitter: 'contact', type: 'data_record' } }, // depth 1
+              { id: 'r2', key: 'parent', value: { submitter: 'contact', type: 'data_record' } }, // depth 0
+              { id: 'r3', key: 'contact', value: { submitter: 'contact', type: 'data_record' } }, // depth 1
+              { id: 'r4', key: 'place', value: { submitter: 'p', type: 'data_record' } }, // depth 1
+              { id: 'r5', key: 'place', value: { submitter: 'contact', type: 'data_record' } }, // depth 1
+            ]
+        });
+
+      return service
+        .getAllowedDocIds({
+          subjectIds,
+          userCtx: { name: 'user', facility_id: 'parent', contact_id: 'contact' },
+          contactDepth: 1,
+          reportDepth: 0,
+          subjectsDepth: { 'parent': 0, 'contact': 1, 'place': 1 },
+        })
+        .then(result => {
+          result.should.have.members([
+            '_design/medic-client', 'org.couchdb.user:user',
+            'r1', 'r2', 'r3', 'r5'
+          ]);
+        });
+    });
+  });
+
+  describe('allowedDepth', () => {
+    // todo
```

Comment:
TODO?

Reply:
****

Reply:
Ah! I see what happened here! I added this so I wouldn't forget, then renamed the function to `isAllowedDepth` and added tests for that one https://github.com/medic/cht-core/blob/6352-adds-report-replication-depth/api/tests/mocha/services/authorization.spec.js#L2910 and overlooked this old todo.